luigi/contrib/external_program.py

```diff
@@ -180,7 +183,9 @@ def _track_url_by_pattern():
                         file_to_write.write(new_line)
                     match = re.search(pattern, new_line.decode('utf-8'))
                     if match:
-                        self.set_tracking_url(match.group(1))
+                        self.set_tracking_url(
+                            self.logs_output_pattern_to_url(match.group(1))
```

Comment:
i don't see how this has any different impact. you're just returning the same value you're passing.

Reply:
@dlstadther
Basically current behaviour is the same, but if I need to change logic of line found in logs transforming I can redefine logs_output_pattern_to_url in my inheritor class

Reply:
@dlstadther

Reply:
Ok makes sense now. I'm only now struggling with the name of the method; it doesn't seem clear right now. Thoughts on `format_tracking_url()` or something like that?

Reply:
agree, name is weird
maybe `convert_to_url()` ?

Reply:
specifically, it's for converting something to a tracking_url

Reply:
then i can suggest

- `convert_to_tracking_url()`
- `make_tracking_url()`
- `make_tracking_url()`
- `tracking_url_from()`

Reply:
@dlstadther 
@honnix 

what do you think ?

Reply:
I will throw one more option. :D

`extract_tracking_url`

Reply:
I'm not sure that's about exctracting _url_
that's more about matching _sth_  from logs into url

Reply:
Isn't the matching part done before calling this method? To me this method will further look into the string and extract `tracking_url`.

What do you have in mind to do in this method exactly? Maybe just call it `make/build_tracking_url`?

Reply:
yes, build is a perfect match